### ISSUE TITLE
Coupon usage not updating

### ISSUE THREAD

**Issue by studiovlijmscherp** [damcclean/simple-commerce#770]
### Description

Updated from 2.4 to 4.x. All is working fine, except the coupon usage counter. When comparing my code/fils with yours at the 4.x branch. It seems i'm missing some controllers. I just have one:
`site/app/Http/Controllers/Controller.php`

You seems to have 8(!). Also; my Middleware folder contains 7 files, you just have one. This makes me feel i might be missing other files as well.. Installed and updated using Composer. And, as said - all seems to be working. Except coupon usage.

### Steps to reproduce

Run Composer install and/or update

### Environment

Environment
Application Name: BeefsAndBeers
Laravel Version: 9.45.1
PHP Version: 8.2.0
Composer Version: -
Environment: local
Debug Mode: ENABLED
URL: beefsandbeers.wip
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp_mailgun
Queue: sync
Session: file

Simple Commerce
Gateways: Mollie
Repository: Customer: DoubleThreeDigital\SimpleCommerce\Customers\EntryCustomerRepository
Repository: Order: DoubleThreeDigital\SimpleCommerce\Orders\EntryOrderRepository
Repository: Product: DoubleThreeDigital\SimpleCommerce\Products\EntryProductRepository
Shipping Methods: Gratis afhalen in Hellendoorn, Bezorging binnen Twente, Bezorging buiten Twente (alleen zaterdagen in oneven weken)
Tax Engine: DoubleThreeDigital\SimpleCommerce\Tax\BasicTaxEngine

Statamic
Addons: 4
Antlers: regex
Stache Watcher: Enabled
Static Caching: Disabled
Version: 3.3.63 Solo

Statamic Addons
cnj/seotamic: 2.1.0
doublethreedigital/duplicator: 2.3.4
doublethreedigital/simple-commerce: 4.2.4
pecotamic/sitemap: 1.4.0

**Comment by duncanmcclean** [comment]
You don't need any controllers inside your app for Simple Commerce to work (or add to any of the middleware) so that's not the issue.

Regarding the coupon usage issue: I'm guessing you're checking it after checking out, correct?

**Comment by studiovlijmscherp** [comment]
👍🏻 I'm checking after successful payment

**Comment by duncanmcclean** [comment]
Cool, I'll take a look when I can.

**Comment by duncanmcclean** [comment]
Sorry, I'm unable to reproduce this issue, even when using Mollie. The stock counter is updated after checkout.

Video: https://www.loom.com/share/15fc3f0f8ca7495eac75f82748f23a7a

A couple of questions:

* Are there any errors in your log file that happen during the checkout process?
* Are you using standard or variant products?
* What is the stock counter at before checking out?

I tried to get your repo up and running (looks like I still have access) and I couldn't manage to get it working with Valet, even after changing things. Your repo seems to be structured differently & missing some things from other Statamic sites which might have been what was causing the issues (everything's inside a `site` directory, no cache folders, `public_html` instead of `public`). I eventually gave up and recorded the video instead - hopefully it's an easy fix!

**Comment by duncanmcclean** [comment]
Sorry, I'm unable to reproduce this issue, even when using Mollie. The 'Redeemed' counter is updated after checkout.

Video: https://www.loom.com/share/ee385da6b30a464486f80747e1606482

A couple of questions:

* Are there any errors in your log file that happen during the checkout process?
* What is the coupon counter at before checking out?

I tried to get your repo up and running (looks like I still have access) and I couldn't manage to get it working with Valet, even after changing things. Your repo seems to be structured differently & missing some things from other Statamic sites which might have been what was causing the issues (everything's inside a site directory, no cache folders, public_html instead of public). I eventually gave up and recorded the video instead - hopefully it's an easy fix!
